### PR TITLE
fix reindex native

### PR DESCRIPTION
**Pull request by calellowitz**
## Technical Summary

This fixes a couple bugs left over from the refactor in https://github.com/dimagi/commcare-hq/pull/32370

### Safety story

The replacement functions all have the exact same behavior, they were just being passed incorrect arguments, or assumed an incorrect return value.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/hqcase/management/commands/reindex_es_native.py [hunk 1]
```diff
@@ -117,16 +117,6 @@ def _initialize_target(es, target_index_info):
...
-            "index": source_index,
-        },
-        "dest": {
-            "index": target_index
-        },
-        "conflicts": "proceed"
-    }
-
     result = manager.reindex(source_index, dest_index, wait_for_completion=False)
     return task_id
```
Comment (dannyroberts) [inline corehq/apps/hqcase/management/commands/reindex_es_native.py:121]: Huh, do you know what's up with this warning?

Comment (calellowitz) [reply corehq/apps/hqcase/management/commands/reindex_es_native.py]: yep, i used the wrong variable name. fixed it and force pushed to clean up the history a bit

**Review by AmitPhulera — APPROVED** [review]
Whoops! I guess I refactored that in a hurry 😅 
Thanks for the fix @calellowitz ⭐

File: corehq/apps/hqcase/management/commands/reindex_es_native.py [hunk 2]
```diff
@@ -117,19 +117,8 @@ def _initialize_target(es, target_index_info):
...
-            "index": target_index
-        },
-        "conflicts": "proceed"
-    }
-
-    result = manager.reindex(reindex_query, wait_for_completion=False)
-    task_id = result["task"]
-    return task_id
+    result = manager.reindex(source_index, target_index, wait_for_completion=False)
+    return result
```
Comment (millerdev) [inline corehq/apps/hqcase/management/commands/reindex_es_native.py:121]: nit
```suggestion
    return manager.reindex(source_index, target_index, wait_for_completion=False)
```

**Review by millerdev — APPROVED** [review]

**Review by snopoke — APPROVED** [review]

**Pull request merged**